myaws/ssm_parameter_env.go

```diff
@@ -15,21 +15,37 @@ type SSMParameterEnvOptions struct {
 
 // SSMParameterEnv prints SSM parameters as a list of environment variables.
 func (client *Client) SSMParameterEnv(options SSMParameterEnvOptions) error {
-	metadata, err := client.FindSSMParameterMetadata(options.Name)
-	if err != nil {
-		return err
-	}
+	var parameters []*ssm.Parameter
+	// Since GetSSMParameters does not have a hierarchy, it is necessary to retrieve all keys at first, then filter the target keys.
+	// Use the DescribeParameters API when retrieving the list of keys.
+	// The rate limit of the DescribeParameters API is not publicly available.
+	// When your SSM Parameter store have large number of keys, it is possible to exceed the rate limit.
+	// https://docs.aws.amazon.com/systems-manager/latest/APIReference/API_DescribeParameters.html
+	// The GetParametersByPath API has a hierarchy separated by '/'.
+	// This API has less impact to the rate limit.
+	// https://docs.aws.amazon.com/systems-manager/latest/APIReference/API_GetParametersByPath.html
```

Comment:
```suggestion
	// Since GetSSMParameters does not have a hierarchy, it is necessary to
	// retrieve all keys at first, then filter the target keys. To do this, we
	// need to call the DescribeParameters API multiple times, but its rate limit
	// seems to be quite low and undocumented.
	// https://docs.aws.amazon.com/systems-manager/latest/APIReference/API_DescribeParameters.html
	// This causes a rate limit exception if your SSM Parameter store have large
	// number of keys.
	// https://github.com/minamijoyo/myaws/issues/31
	//
	// The GetParametersByPath API has a hierarchy separated by '/'. This means
	// it has less impact to the rate limit.
	// https://docs.aws.amazon.com/systems-manager/latest/APIReference/API_GetParametersByPath.html
	//
	// So we use it if the path seems to have a hierarchy and fall back to
	// the original behavior if not.
```

Reply:
I rewrote it a bit 😎

Reply:
Thanks! It looks good for me.